lm_eval/tasks/hendrycks_test.py

```diff
@@ -0,0 +1,137 @@
+            'formal_logic', 'global_facts', 'high_school_biology', 'high_school_chemistry', 'high_school_computer_science', 
+            'high_school_european_history', 'high_school_geography', 'high_school_government_and_politics', 'high_school_macroeconomics', 
+            'high_school_mathematics', 'high_school_microeconomics', 'high_school_physics', 'high_school_psychology', 'high_school_statistics', 
+            'high_school_us_history', 'high_school_world_history', 'human_aging', 'human_sexuality', 'international_law', 'jurisprudence', 
+            'logical_fallacies', 'machine_learning', 'management', 'marketing', 'medical_genetics', 'miscellaneous', 'moral_disputes', 
+            'moral_scenarios', 'nutrition', 'philosophy', 'prehistory', 'professional_accounting', 'professional_law', 'professional_medicine', 
+            'professional_psychology', 'public_relations', 'security_studies', 'sociology', 'us_foreign_policy', 'virology', 'world_religions']
+
+CHOICES = ['A','B','C','D']
+
+def create_all_tasks():
+    """Creates a dictionary of tasks from a list of subjects
+    :return: {task_name: task}
+        e.g. {hendrycksTest-abstract_algebra: Task, hendrycksTest-anatomy: Task}
+    """
+    return {
+        f"hendrycksTest-{sub}": create_task(sub) for sub in SUBJECTS
+    }
+
+def create_task(subject):
+    class HendrycksTest(GeneralHendrycksTest):
+        def __init__(self):
+            super().__init__(subject)
+    return HendrycksTest
+
+class GeneralHendrycksTest(MultipleChoiceTask):
+
+    def __init__(self, subject):
+        self.subject = subject
+        super().__init__()
+
+    def download(self):
+        
+        self.data_dir = "data/hendrycksTest/"
+        if not os.path.exists(self.data_dir):
+            sh("""
+                mkdir -p data
+                wget https://people.eecs.berkeley.edu/~hendrycks/data.tar -P data/
+                tar -xf data/data.tar -C data/
+                rm data/data.tar
+                mv data/data data/hendrycksTest
+                """)
+
+    def has_training_docs(self):
+        return True
+
+    def has_validation_docs(self):
+        return False
+
+    def has_test_docs(self):
+        return True
+
+    def _load_docs(self, filename):
+
+        reader = csv.reader(open(filename, 'r'), quotechar='"', delimiter=',')
+
+        docs = []
+        for row in reader:
+            doc = {
+                "query": self._format_example(row),
+                "choices": CHOICES,
+                "gold": CHOICES.index(row[5])
+            }
+            docs.append(doc)
+        return docs
+
+    def _format_example(self, row):
+        """
+            <prompt>
+            A. <choice1>
+            B. <choice2>
+            C. <choice3>
+            D. <choice4>
+            Answer:
+        """
+        prompt = row[0]
+        for j in range(4):
+            prompt += "\n{}. {}".format(CHOICES[j], row[j+1])
+        prompt += "\nAnswer:"
+        return prompt
```

Comment:
The answers should not be included in the context. Instead, the format should look like "Question: <prompt>\nAnswer:"

Reply:
Some problems are specified with the answer choices. For example,
```
A 63-year-old man is brought to the emergency department because of a 4-day history of increasingly severe left leg pain and swelling of his left calf. He also has a 1-month history of
increasingly severe upper midthoracic back pain. During this time, he has had a 9-kg (20-lb)
weight loss despite no change in appetite. He has no history of major medical illness. His only
medication is ibuprofen. He is 180 cm (5 ft 11 in) tall and weighs 82 kg (180 lb); BMI is 25
kg/m2 . His vital signs are within normal limits. On examination, lower extremity pulses are
palpable bilaterally. The remainder of the physical examination shows no abnormalities. An x-ray
of the thoracic spine shows no abnormalities. A CT scan of the abdomen shows a 3-cm mass in
the body of the pancreas; there are liver metastases and encasement of the superior mesenteric
artery. Ultrasonography of the left lower extremity shows a femoropopliteal venous clot. Which
of the following is the most likely cause of this patient’s symptoms?
(A) Carcinoid syndrome
(B) Hypercoagulability from advanced malignancy
(C) Multiple endocrine neoplasia
(D) Splenic artery aneurysm and embolic disease of the left lower extremity
```
requires looking at all options to find the "most likely" option.